docs/en/Community-Articles/2024-05-09-what-is-object-to-object-mapping/post.md

```diff
@@ -0,0 +1,122 @@
+# What is Object to Object Mapping
+
+Mapping the properties of one object to the properties of another object is called Object to object mapping. Most of the time we don't want to show the data we store to end users as it is. We only give users the information they will use for that operation. In tables that contain relationships, we analyze the relationships and present meaningful data to users. For example, suppose we have a product and a category object, we keep an attribute called categoryId in the product object. However, it would be illogical to show the categoryId attribute to users. Therefore, we create dto's (data transfer objects) and show the category name attribute to users instead of the category Id attribute. 
+
+DTOs are used to transfer data of objects from one place to another. We often need to convert our Domain objects to DTOs and DTOs to Domain objects, for this you can use a code like this. 
+
+````csharp
+    public virtual async Task<CustomerDto> CreateAsync(CustomerCreateDto input)
+    {
+
+        var customer = await _customerManager.CreateAsync(
+        input.BirthDay, input.MembershipDate, input.FirstName, input.LastName
+        );
+        CustomerDto customerDto = new CustomerDto
+        {
+            Id = customer.Id,
+            FirstName = input.FirstName,
+            LastName = input.LastName,
+            // ...other
+        };
+        
+        return customerDto;
+    }
+
+````
+This code is repetitive, tedious to write. It violates DRY principles, makes your code very complex and reduces readability.  Instead of using this approach, you can use the automapper which does the matching automatically. Here is a better code:
+
+
+````csharp
+        public virtual async Task<CustomerDto> CreateAsync(CustomerCreateDto input)
+        {
+
+            var customer = await _customerManager.CreateAsync(
+            input.BirthDay, input.MembershipDate, input.FirstName, input.LastName
+            );
+
+            return ObjectMapper.Map<Customer, CustomerDto>(customer);
+        }
+````
+
```

Comment:
You should write a small explanation before/after each code block in articles. For example, in here, you are using the `ObjectMapper.Map` method and you can mention that 

> "Whenever you create a new abp-based application, the related mapping configurations are automatically done in your template and you can directly use it like in the example above. Also note that, `ObjectMapper` is defined in the `ApplicationService` base class, so you don't need to manually inject the `IObjectMapper` interface."

And then you can link the sentence with the following section with a sentence like: "Let's first see what AutoMapper library is used for and see it in action..."

> Please use your own sentences here, I just write them to give you a idea.

Reply:
I added this sentence:
The **Map** method allows you to convert your `Customer` entity to `CustomerDto`. Let's learn more about **AutoMapper**